index/scorch/persister.go

```diff
@@ -633,14 +633,19 @@ func (s *Scorch) removeOldBoltSnapshots() (numRemoved int, err error) {
 		return 0, err
 	}
 
-	if len(persistedEpochs) <= NumSnapshotsToKeep {
+	numSnapshotsToKeep := NumSnapshotsToKeep
+	if val, ok := s.config["numSnapshotsToKeep"].(float64); ok && val > 0 {
```

Comment:
i think this keeps coming up from time to time.  from the Go library perspective this is kinda ugly as users who pass an integer (what you would expect) find that the feature doesn't work.  i think if we add these kind of options, they should make sense from the Go library perspective first, even if that means more work elsewhere to parse JSON differently...